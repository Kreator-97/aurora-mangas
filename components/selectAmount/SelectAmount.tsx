import { FC, useState } from 'react'
import { HiOutlineMinusCircle, HiOutlinePlusCircle } from 'react-icons/hi'

interface Props {
  initial: number;
  id?: string;
  minValue?: number;
  maxValue?: number;
  onChangeAmount?: (amount:number, id:string) => void;
}

export const SelectAmount:FC<Props> = ({initial, id = '', onChangeAmount, minValue = 1, maxValue = Infinity}) => {
  const [ counter, setCounter] = useState(initial)

  const onChangeAmountEvent = (value: number, type: 'decrement' | 'increment') => {
    if( type === 'decrement' ) {
      const newValue = Math.max(minValue, counter-value)
      setCounter(newValue)
      onChangeAmount && onChangeAmount(newValue, id)
      return
    }
    
    if( type === 'increment' ) {
      const newValue = Math.min(maxValue, counter+value)
      setCounter(newValue)
      onChangeAmount && onChangeAmount(newValue, id)
      return
    }
  }

  return (
    <div>
      <h2 className='text-lg mb-2 text-center'>Seleccione la cantidad</h2>

      <div className='flex gap-4 mb-2 justify-center'>
        <div>
          <HiOutlineMinusCircle
            size={28}
            cursor="pointer"
            onClick={() => onChangeAmountEvent(1, 'decrement') }
            data-testid={'decrement-icon'}
            strokeWidth={2}
          />
        </div>
        <div
          className='font-medium text-2xl px-2'
          data-testid={'amount-box'}
        >{ counter }
        </div>
        <div>
          <HiOutlinePlusCircle
            size={28}
            cursor="pointer"
            onClick={() => onChangeAmountEvent(1, 'increment') }
            data-testid={'increment-icon'}
          />
        </div>
      </div>
    </div>
  )
}
