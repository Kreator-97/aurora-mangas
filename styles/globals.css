@tailwind base;
@tailwind components;
@tailwind utilities;
@tailwind variants;

:root {
  box-sizing: border-box;
  --bg-color: #001E31;
  --bg-color2: #00474F;
  --stroke-color: #07695b70;
  --success: #21C728;
  --success-dark: #118A16;
  --dark: #333;
  --light: #eee;
  --header-height: 88px;
  --sm: 480px;
  --md: 720px;
  --lg: 960px;
  --xl: 1200px;
  --max: 1440px;
}

html {
  color: var(--light);
  background-image: linear-gradient( var(--bg-color), var(--bg-color2));
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  background-attachment: fixed;
}

.main {
  min-height: calc(100vh -var(--header-height));
  margin: var(--header-height) auto 0;
}

.title {
  @apply text-xl md:text-2xl lg:text-3xl text-center py-2;
}

.btn {
  @apply px-2 py-1 font-normal text-light rounded transition-colors border border-solid border-light;
}

.btn.ghost {
  @apply text-accent bg-opacity-0 border-accent;
}

.bg-success-gradient {
  background-image: linear-gradient( var(--success), var(--success-dark))!important;
}

.btn.ghost:hover {
  @apply bg-light;
}

.active {
  @apply bg-success;
}

@media screen and (min-width: 720px) {
  .main {
    padding: 0 .5rem;
  }
}
