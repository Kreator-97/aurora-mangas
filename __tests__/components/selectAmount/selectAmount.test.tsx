import { expect } from '@jest/globals'
import { fireEvent, render, screen } from '@testing-library/react'
import { SelectAmount } from '../../../components/selectAmount/SelectAmount'

describe('tests on SelectAmount component', () => {
  test('should to match with snapshot', () => {
    const { container } = render(<SelectAmount initial={1} />)
    const title = screen.getByText('Seleccione la cantidad')
    const amount = screen.getByText('1')

    expect(container).toMatchSnapshot()
    expect(title)
    expect(amount)
  })

  test('should to call onChangeAmount event', () => {
    const onChangeAmountMock = jest.fn()
    const id = '123-abc-xyz'
    render(
      <SelectAmount
        initial={1}
        onChangeAmount={ onChangeAmountMock }
        minValue={1}
        id={id}
      />
    )

    const incrementIcon = screen.getByTestId('increment-icon')
    const decrementIcon = screen.getByTestId('decrement-icon')

    fireEvent.click(incrementIcon)
    expect(onChangeAmountMock).toHaveBeenCalledWith(2, id)

    const amount = screen.getByTestId('amount-box')
    expect( amount.textContent ).toBe('2')

    fireEvent.click(decrementIcon)
    expect(onChangeAmountMock).toHaveBeenCalledWith(1, id)

    expect( amount.textContent ).toBe('1')
  })
})
