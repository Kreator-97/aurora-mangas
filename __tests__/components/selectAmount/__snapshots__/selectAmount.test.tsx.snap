// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on SelectAmount component should to match with snapshot 1`] = `
<div>
  <div>
    <h2
      class="text-lg mb-2 text-center"
    >
      Seleccione la cantidad
    </h2>
    <div
      class="flex gap-4 mb-2 justify-center"
    >
      <div>
        <svg
          cursor="pointer"
          data-testid="decrement-icon"
          fill="none"
          height="28"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="28"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15 12H9m12 0a9 9 0 11-18 0 9 9 0 0118 0z"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </div>
      <div
        class="font-medium text-2xl px-2"
        data-testid="amount-box"
      >
        1
      </div>
      <div>
        <svg
          cursor="pointer"
          data-testid="increment-icon"
          fill="none"
          height="28"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="28"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 9v3m0 0v3m0-3h3m-3 0H9m12 0a9 9 0 11-18 0 9 9 0 0118 0z"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
